### ISSUE TITLE
Unable to resolve module zlib from: ...

### ISSUE THREAD

**Issue by BesartH** [expo/expo#17302]
Hello I am trying to use Expo Server SDK Node for sending push notifications but I am getting this error:

```
Unable to resolve module zlib from /Users/b/Desktop/vivelo-app/node_modules/expo-server-sdk/build/ExpoClient.js: zlib could not be found within the project or in these directories:
  node_modules/expo-server-sdk/node_modules
  node_modules
```

```
If you are sure the module exists, try these steps:
 1. Clear watchman watches: watchman watch-del-all
 2. Delete node_modules and run yarn install
 3. Reset Metro's cache: yarn start --reset-cache
 4. Remove the cache: rm -rf /tmp/metro-*
  20 | const node_fetch_1 = require("node-fetch");
  21 | const promiseLimit = require("promise-limit");
> 22 | const zlib = require("zlib");
| const BASE_URL = 'https://exp.host';
  24 | const BASE_API_URL = `${BASE_URL}/--/api/v2`;
```

**Comment by byCedric** [comment]
`zlib` is a Node core module, you can't use it in React Native. `expo-server-sdk` should only be used from a (node) server. If you need to interact with the notifications API, please see https://docs.expo.dev/versions/latest/sdk/notifications/

**Comment by BesartH** [comment]
> `zlib` is a Node core module, you can't use it in React Native. `expo-server-sdk` should only be used from a (node) server. If you need to interact with the notifications API, please see https://docs.expo.dev/versions/latest/sdk/notifications/

Thanks for making this clear.

**Comment by Dan-CB** [comment]
Hey guys I'm struggling with this one at the moment. I'm not sure what exactly I should look for in the linked API docs and I'd preferably want to try to get the package to work. Is it just in Expo Go that this has issues (where I'm having the error on load) or will it also happen on build? Sorry, I just can't find any other places to look/ask.

**Comment by yew12** [comment]
> Hey guys I'm struggling with this one at the moment. I'm not sure what exactly I should look for in the linked API docs and I'd preferably want to try to get the package to work. Is it just in Expo Go that this has issues (where I'm having the error on load) or will it also happen on build? Sorry, I just can't find any other places to look/ask.

I also have this same issue and question if anyone has an answer to this. Thanks.

**Comment by Dan-CB** [comment]
I've since made sense of it I think. I was trying to use the package inside my react native app to try to call server notifications from the app. I don't really understand the technicalities of it and core node libraries not being present in a RN project, but essentially you just cannot use the package inside the app code. It HAS to be a separate server/lambda (if using AWS for instance) that exists outside of the app code. In my use case, I will set up a lambda function on AWS that will host this functionality and whatever ends up triggering it, the package should in theory work as intended. Hope that helps. I'm still a n00b mind you guys so will see when I get around to it.

**Comment by yew12** [comment]
To follow up on that, I was able to get it to work as well. I was getting the issue when trying to use expo-server-sdk-node and I couldn't use the node library with react native. I downloaded "node-libs-react-native": "^1.2.1" and added "import "node-libs-react-native/globals.js";" in my root file and everything seems to be working fine now. I am not sure how this will effect the code when in production though.
